samza-core/src/test/java/org/apache/samza/runtime/TestClusterBasedProcessorLifecycleListener.java

```diff
@@ -79,11 +79,14 @@ public void testLifecycleListenerAfterFailure() {
   public void testShutdownHookInvokesShutdownHookCallback() {
     doNothing().when(mockShutdownHookCallback).run();
 
-    // setup shutdown hook
+    // call beforeStart to setup shutdownHook
     clusterBasedProcessorLifecycleListener.beforeStart();
 
     // Simulating shutdown hook invocation by JVM
     clusterBasedProcessorLifecycleListener.getShutdownHookThread().run();
+
+    // call afterStop to countdown shutdownLatch
+    clusterBasedProcessorLifecycleListener.afterStop();
```

Comment:
Just to clarify, this is to address the following line always waiting until the timeout? https://github.com/apache/samza/blob/9b3a9e68bca8cfef8a87542143212015b5263cea/samza-core/src/main/java/org/apache/samza/runtime/ClusterBasedProcessorLifecycleListener.java#L95

Reply:
Yes, that is correct. Sorry I didn't make it clear.